spec/support/builders.rb

```diff
@@ -368,6 +368,9 @@ def build_repo(path, &blk)
     end
 
     def update_repo(path)
+      if path == gem_repo1 && caller_locations.first.label != "build_repo"
+        raise "Updating gem_repo1 is unsupported -- use gem_repo2 instead"
```

Comment:
How about "It's not possible to update gem_repo1 because that would cause side effects in every other test. Use gem_repo2 instead, which is rebuilt (if needed) for each test."